src/libs/kata-types/src/k8s.rs

```diff
@@ -77,10 +96,119 @@ pub fn container_type_with_id(spec: &oci::Spec) -> (ContainerType, Option<String
     (container_type, sid)
 }
 
+// count_files will return the number of files within a given path.
+// If the total number of
+// files observed is greater than limit, break and return -1
+fn count_files<P: AsRef<Path>>(path: P, limit: i32) -> std::io::Result<i32> {
+    // First, Check to see if the path exists
+    let src = std::fs::canonicalize(path)?;
+
+    // Special case if this is just a file, not a directory:
+    if !src.is_dir() {
+        return Ok(1);
+    }
+
+    let mut num_files = 0;
+
+    for entry in std::fs::read_dir(src)? {
+        let file = entry?;
+        let p = file.path();
+        if p.is_dir() {
+            num_files += count_files(&p, limit)?;
+        } else {
+            num_files += 1;
+        }
+
+        if num_files > limit {
+            return Ok(-1);
+        }
+    }
+
+    return Ok(num_files);
+}
+
+/// Check if a volume should be processed as a watchable volume,
+/// which adds inotify-like function for virtio-fs.
+pub fn is_watchable_mount<P: AsRef<Path>>(path: P) -> bool {
+    if !is_secret(&path) && !is_configmap(&path) {
+        return false;
+    }
+
+    // we have a cap on number of FDs which can be present in mount
+    // to determine if watchable. A similar Check exists within the agent,
+    // which may or may not help handle case where extra files are added to
+    // a mount after the fact
+    let count = count_files(&path, 8).unwrap_or(0);
```

Comment:
Can you explain why the limit is set to 8, and maybe defining it as const is better.

Reply:
The original PR is https://github.com/kata-containers/kata-containers/pull/1964, the design is compatible with the golang runtime and agent. So here used 8 and the driver is `watchable-bind`.